packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -284,7 +284,14 @@ export function clean_nodes(
 					(attribute) => attribute.type === 'Attribute' && attribute.name.startsWith('--')
 				)));
 
-	return { hoisted, trimmed, is_standalone };
+	// if a component or snippet starts with text, we need to add an anchor comment
+	// so that its text node doesn't get fused with its surroundings
```

Comment:
It doesn't — the annotation only travels with the value within the same scope, unless it's on the return value...

```js
return {
  hoisted,
  trimmed,
  is_standalone,
  /** if a component or snippet starts with text, we need to add an anchor comment so that its text node doesn't get fused with its surroundings */
  is_text_first
};
```

...but we could certainly do that instead